src/server/controllers/user.js

```diff
@@ -4,33 +4,48 @@ const { PrismaClient } = require('@prisma/client');
 const prisma = new PrismaClient();
 
 const jwtSecret = 'mysecret';
+const saltRounds = 10;
 
 const register = async (req, res) => {
     const { username, password } = req.body;
 
-    const createdUser = null;
-
-    res.json({ data: createdUser });
+    bcrypt.hash(password, saltRounds, async (err, hash) => {
+        const newUser = await prisma.user.create({
+            data: {
+                username,
+                password: hash
+            },
+            select: {
+                username: true
+            }
+        })
+
+        res.status(201).json({ user: newUser })
+    });
 };
 
 const login = async (req, res) => {
     const { username, password } = req.body;
-
-    const foundUser = null;
-
-    if (!foundUser) {
-        return res.status(401).json({ error: 'Invalid username or password.' });
+    let token = false
+    const foundUser = await prisma.user.findUnique({
+        where: {
+            username: username
+        }
+    })
+
+    if(foundUser) {
+        bcrypt.compare(password, foundUser.password, async (err, result) => {
+            if(result) {
+                token = jwt.sign({username}, jwtSecret)
+                res.status(201).send({ token })
+            } else {
+                res.status(401).send('Invalid username or password')
+            }
+        })
+    } else {
+        res.status(401).send('Invalid username or password')
```

Comment:
our code looks almost identical xD good job!

Reply:
> our code looks almost identical xD good job!

I checked yours and our code is literally the exact same with some minor differences haha, great minds think alike!